src/main/java/frc/robot/subsystems/BallPath/Shooter/ShooterImpl.java

```diff
@@ -38,11 +103,233 @@ public int checkBalls(){
     }
 
     @Override
-    public void setHoodAngle(double distance){}
+    public void setHoodAngle(double setPointHood){
+        if(turretHoodPosition >= setPointHood - hoodBuffer && turretHoodPosition <= setPointHood + hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, 0);
+          hoodReady = true;
+        }else if(turretHoodPosition <= setPointHood - hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, hoodSpeed);
+          hoodReady = false;
+        }else if (turretHoodPosition >= setPointHood + hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, -hoodSpeed);
+          hoodReady = false;
+        }
+    }
 
     @Override
     public void start(){}
 
     @Override
-    public void stop(){}
+    public void stop(){
+        System.out.println("DISABLING THE SHOOTER");
+        setPointHood = 0;
+        setPointRotation = 0;
+
+        shooterMotor.set(ControlMode.PercentOutput, 0);
+
+        if(turretEncoderReadingPosition >= setPointRotation - turretBuffer && turretEncoderReadingPosition <= setPointRotation + turretBuffer){
+          turretMotor.set(ControlMode.PercentOutput, 0);
+          turretReady = true;
+        }else if(turretEncoderReadingPosition <= setPointRotation - turretBuffer){
+          turretMotor.set(ControlMode.PercentOutput, turretSpeed);
+          turretReady = false;
+        }else if (turretEncoderReadingPosition >= setPointRotation + turretBuffer){
+          turretMotor.set(ControlMode.PercentOutput, -turretSpeed);
+          turretReady = false;
+        }
+        
+        if(turretHoodPosition >= setPointHood - hoodBuffer && turretHoodPosition <= setPointHood + hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, 0);
+          hoodReady = true;
+        }else if(turretHoodPosition <= setPointHood - hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, hoodSpeed);
+          hoodReady = false;
+        }else if (turretHoodPosition >= setPointHood + hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, -hoodSpeed);
+          hoodReady = false;
+        }
+
+    }
+
+    @Override
+    public void shootFender() {
+        // Fender shot for teleop
+        System.out.println("FENDER SHOT: ");
+        setPointHood = 100000;
+        setPointShooterPID = 0; // tbd
+        setPointRotation = 0;
+        System.out.println("SETPOINT HOOD: " + setPointHood);
+        System.out.println("SETPOINT SHOOTER " + setPointShooter);
+        System.out.println("SETPOINT FOR THE TURRET TURNING " + setPointRotation);
+
+        // pid for shooter
+        if(setPointShooterPID != 0){
+          currentOutput = shooterPid.calculate(shooterEncoderReadingVelocity, setPointShooter);
```

Comment:
Here, for example, the `setPointShooterPID` is checked for being non-zero, but then the `setPointShooter` variable is the one used to calculate the PID loop. Seems like a mistake.